packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -361,6 +386,25 @@ export class TaskDefinition extends TaskDefinitionBase {
    * Adds a new container to the task definition.
    */
   public addContainer(id: string, props: ContainerDefinitionOptions) {
+    if (this.cpu !== undefined
+          && props.cpu !== undefined
+          && parseInt(this.cpu, 10) < props.cpu) {
+      this.node.addWarning('CPU specified for the container cannot be greater than the CPU for the task definition');
+    }
+    if (this.memoryMiB !== undefined
+          && props.memoryLimitMiB !== undefined
+          && parseInt(this.memoryMiB, 10) < props.memoryLimitMiB) {
+      this.node.addWarning('Memory specified for the container cannot be greater than the memory for the task definition');
+    }
+    if (this.remainingMemoryMiB !== undefined
+          && props.memoryLimitMiB !== undefined
+          && parseInt(this.remainingMemoryMiB, 10) > (parseInt(this.remainingMemoryMiB, 10) - props.memoryLimitMiB)) {
```

Comment:
It pains me that we're doing integer math on a string :(.

Reply:
Agreed, I wanted to get this change in, and then work on https://github.com/aws/aws-cdk/issues/5088. I wonder if it makes sense to get that change merged in and then merge this PR in?